github/github-release/github-release.yaml

```diff
@@ -0,0 +1,106 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: github-release
+spec:
+  workspaces:
+    - name: upload
+  params:
+    - name: TAG_NAME
+      description: Name of the tag
+      type: string
+    - name: NAME
+      description: Name of the release
+      type: string
+    - name: BODY
+      type: string
+      description: Short description of the release.
+      default: ""
+    - name: OWNER
+      type: string
+      description: User name of the Github owner
+    - name: REPO_NAME
+      type: string
+      description: Full name of the repository where the release is to made.
+    - name: BRANCH
+      type: string
+      description: Branch on which release is to be made.
+      default: master
+    - name: UPLOAD_ASSET_FILE_NAME
+      type: string
+      description: Name of the asset to be uploaded.
+      default: ""
+    - name: github-token-secret
+      type: string
+      description: name of the secret holding the github-token.
+      default: github-secret
+    - name: github-token-secret-key
+      type: string
+      description: name of the secret key holding the github-token.
+      default: api-token
+  steps:
+    - name: release
+      image: registry.access.redhat.com/ubi8/ubi-minimal:latest
+      script: |
+        #!/usr/bin/env bash
```

Comment:
hm.  there's a tool called [`hub`](https://hub.github.com/#scripting) that might be useful here?  It would replace the need to have bash marshaling JSON & making a series of curl requests.

Reply:
We do something similar to this in Tekton's plumbing for releases.  See the Task we use here for an example of utilizing `hub`: https://github.com/tektoncd/plumbing/blob/master/tekton/resources/release/github_release.yaml#L212-L223

Reply:
Okay, will go through this tool. I wasn't aware of this :sweat_smile: 
I guess, this will definitely improve the task. Thanks :+1: